src/RouterWrapper.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from "react";
+import ChrisAPIClient from "./api/chrisapiclient";
+import { AppContext } from "./context/context";
+import { Types } from "./context/actions/types";
+import { IUserState } from "./context/reducers/userReducer";
+import Client, { User } from "@fnndsc/chrisapi";
+import { useLocation, useHistory } from 'react-router-dom'
+
+/**
+ * Allows user to bypass Login page if already authenticated by checking AUTH_TOKEN
+ * in session storage
+ */
+const RouterWrapper: React.FC = ({ children }) => {
+  const { dispatch } = React.useContext(AppContext);
+  const [renderChildren, setRenderChildren] = useState(false);
+  const location = useLocation();
+  const history = useHistory();
+
+  useEffect(() => {
+    const token = window.sessionStorage.getItem("AUTH_TOKEN");
+    (async () => {
+      if (!!token) {
+        const client: Client = ChrisAPIClient.getClient();
+        const res: User = await client.getUser();
+        const user: IUserState = res?.data;
+        dispatch({
+          type: Types.Login_update,
+          payload: user
+        });
+        
+        // If on Login page and already authenticated, go to Dashboard
+        if (location.pathname === "/login") {
+          history.push('/');
+        }
+      }
+      setRenderChildren(true);
```

Comment:
Would we want this inside the `if` statement, since we only want the value to be true, if there's a valid token?

Reply:
If there is no token, we proceed with rendering the children. The reason behind not rendering right away is we needed to ensure tokens were checked before the RouteGuard checks whether the user has already logged in or not.